src/openrct2/Context.cpp

```diff
@@ -1075,14 +1077,19 @@ namespace OpenRCT2
 
             UpdateTimeAccumulators(deltaTime);
 
-            if (useVariableFrame)
+            if (useVariableFrame || gConfigGeneral.MaxFPS > 40)
             {
                 RunVariableFrame(deltaTime);
             }
             else
             {
                 RunFixedFrame(deltaTime);
             }
+
+            if (gConfigGeneral.MaxFPS > 0)
+            {
+                Platform::Sleep(1000 / (gConfigGeneral.MaxFPS * 2));
+            }
```

Comment:
This doesn't look like a good method of achieving this. Ideally, you'd tie in with the existing loop functions, rather than simply blocking the thread for a number of milliseconds.

Reply:
Ok I am refactoring how it works.

Reply:
I have moved it co-op with the existing variable refresh rate function. It works by running at a variable refresh rate but then blocks the thread effectively locking the fps in place. Is that what you meant or should I change it to not block the thread?